contracts/NFTFactory.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract NFTFactory is ERC721Enumerable, Ownable, ERC721Burnable {
```

Comment:
I believe there is no need to inherit from `ERC721Burnable` - we had requirements about `burn()` function. If you inherit from `ERC721Burnable`, then your contract needs to override `_beforeTokenTransfer` and `supportsInterface` functions as you get 2 base contracts (ERC721) define the same functions.
If you really want to have `burn()` function, you can add such function instead of inheriting from `ERC721Burnable`, with a modifier to allow only owner to burn a token.